pkgs/development/ocaml-modules/xenstore_transport/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, buildDunePackage, fetchFromGitHub, xenstore, lwt }:
+
+buildDunePackage rec {
+  pname = "xenstore_transport";
+  version = "1.3.0";
+
+  minimumOCamlVersion = "4.04";
+  useDune2 = true;
+
+  src = fetchFromGitHub {
+    owner = "xapi-project";
+    repo = "ocaml-xenstore-clients";
+    rev = "v${version}";
+    sha256 = "1kxxd9i4qiq98r7sgvl59iq2ni7y6drnv48qj580q5cyiyyc85q3";
+  };
+
+  propagatedBuildInputs = [ xenstore lwt ];
```

Comment:
check this failure: https://github.com/anmonteiro/nix-overlays/runs/1786341313#step:5:1245

Reply:
oh that's actually saying xenstore requires it, not xenstore_transport.

Reply:
hmmm it could also be a dune 2 quirk. I default to dune2 for building all packages

Reply:
I've built all irmin/git-mirage packages which transitively depend on `xenstore` with dune 2, since `xenstore` has `useDune2` set and it suceeded. Really weird. Maybe OCaml 4.11 issue?

Reply:
interesting. Yeah I built this on OCaml 4.11

Reply:
On `master` everything builds fine for me with 4.11 as well, can you reproduce it there?

Reply:
Hm, I triggered it as well now in a `nixpkgs-review` with OCaml 4.10. Weird.

```
builder for '/nix/store/5fxwkbvnmjg61dxi60dsqysxgjg6gmri-ocaml4.10.0-xenstore_transport-1.3.0.drv' failed with exit code 1; last 10 log lines:
  building
  File "/nix/store/1f7xs83sak49nmkkhsbkd8m2bxy1mn1z-ocaml4.10.0-xenstore-2.1.1/lib/ocaml/4.10.0/site-lib/xenstore/dune-package", line 10, characters 19-31:
  10 |  (requires cstruct stdlib-shims)
                          ^^^^^^^^^^^^
  Error: Library "stdlib-shims" not found.
  -> required by library "xenstore" in
     /nix/store/1f7xs83sak49nmkkhsbkd8m2bxy1mn1z-ocaml4.10.0-xenstore-2.1.1/lib/ocaml/4.10.0/site-lib/xenstore
  Hint: try:
    dune external-lib-deps --missing -p xenstore_transport -j 8 @install
```

Reply:
aha! should we add that as a dependency to xenstore then?

Reply:
still a little weird to me that it isn't present upstream, but `dune-package` seems to require it.

Reply:
I'm really confused by this because I don't get what circumstances trigger this issue. On `master` the following turn up no results:

```
grep -r "stdlib-shims" $(nix-build -A ocamlPackages.xenstore --no-out-link)
```

So sometimes `dune-package` requires `stdlib-shims` and sometimes it doesn't?!